scripts/styleChecker.fsx

```diff
@@ -0,0 +1,393 @@
+    let isPrettierPluginXmlInstalled =
+        Process.Execute(
+            {
+                Command = "npm"
+                Arguments = $"list @prettier/plugin-xml@{version}"
+            },
+            Echo.Off
+        )
+        |> IsProcessSuccessful
+
+    if not(isPrettierPluginXmlInstalled) then
+        Process
+            .Execute(
+                {
+                    Command = "npm"
+                    Arguments = $"install @prettier/plugin-xml@{version}"
+                },
+                Echo.Off
+            )
+            .UnwrapDefault()
+        |> ignore
+
+let StyleXamlFiles() =
+    InstallPrettier(prettierVersion)
+    InstallPrettierPluginXml(pluginXmlVersion)
+
+    Process
+        .Execute(
+            {
+                Command = "npm"
+                Arguments =
+                    $"install --save-dev prettier@{prettierVersion} @prettier/plugin-xml@{pluginXmlVersion}"
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+    Process
+        .Execute(
+            {
+                Command = "./node_modules/.bin/prettier"
+                Arguments =
+                    "--xml-whitespace-sensitivity ignore --tab-width 4 --prose-wrap preserve --write '**/*.xaml'"
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+let RunPrettier(arguments: string) =
+
+    // We need this step so we can change the files using `npx prettier --write` in the next step.
+    // Otherwise we get permission denied error in the CI.
+    Process
+        .Execute(
+            {
+                Command = "chmod"
+                Arguments = "777 -R ."
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+    Process.Execute(
+        {
+            Command = "npx"
+            Arguments = $"prettier {arguments}"
+        },
+        Echo.Off
+    )
+    |> UnwrapProcessResult "" true
+    |> ignore
+
+
+    // Since after installing commitlint dependencies package.json file changes, we need to
+    // run the following command to ignore package.json file
+    Process
+        .Execute(
+            {
+                Command = "git"
+                Arguments = "restore package.json"
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+let StyleTypeScriptFiles() =
+    RunPrettier "--quote-props=consistent --write ./**/*.ts"
+
+let StyleYmlFiles() =
+    RunPrettier "--quote-props=consistent --write ./**/*.yml"
+
+let ContainsFiles (rootDir: DirectoryInfo) (searchPattern: string) =
+    Helpers.GetFiles rootDir searchPattern |> Seq.length > 0
+
+let GitDiff() : ProcessResult =
+
+    // Since we changed file modes in the prettier step we need the following command to
+    // make git ignore mode changes in files and doesn't include them in the git diff command.
+    Process
+        .Execute(
+            {
+                Command = "git"
```

Comment:
@realmarv let's do this also only when run in CI (so check for github env var)

Reply:
@realmarv actually this whole function shouldn't be called if not in CI

Reply:
> @realmarv actually this whole function shouldn't be called if not in CI
why?

Reply:
I'm going to make a suggestion for this script and this way you'll understand why: let's split this script in too, one of them should be called styleApply.fsx and the other styleCheck.fsx; the latter should call the former, and only the latter should call git diff (and CI should call the latter)

Reply:
Should I add the functions in this script to the FileConventions library and use them in these two scripts? because in the style check part of the script, I'm calling style functions separately to print a proper error message when each case happens

Reply:
sounds good, this way there's no need for styleCheck.fsx to call styleApply.fsx

Reply:
not sure if I should add tests for those functions